src/neo/Trie/MPT/MPTTrie.cs

```diff
@@ -0,0 +1,247 @@
+using Neo.IO.Json;
+using System;
+
+namespace Neo.Trie.MPT
+{
+    public class MPTTrie : MPTReadOnlyTrie, ITrie
+    {
+        private MPTDb db;
+
+        public MPTTrie(byte[] root, IKVStore store) : base(root, store)
+        {
+            this.db = new MPTDb(store);
+        }
+
+        public bool Put(byte[] path, byte[] value)
+        {
+            path = path.ToNibbles();
+            if (value.Length == 0)
+            {
+                return TryDelete(ref root, path);
+            }
+            var n = new ValueNode(value);
+            return Put(ref root, path, n);
+        }
+
+        private bool Put(ref MPTNode node, byte[] path, MPTNode val)
```

Comment:
When the `path` must be in `nibbles` I think that we should call it `nibblesPath` or do the conversion here

Reply:
I change `path` to `key` and the `nibbles` called `path`.